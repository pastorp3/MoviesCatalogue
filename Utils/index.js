import { checkPropTypes } from 'prop-types';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import RootReducer from '../src/reducers/index';

export const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return component;
};

export const checkProps = (component, expectedProps) => {
  const propsErr = checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
  return propsErr;
}
