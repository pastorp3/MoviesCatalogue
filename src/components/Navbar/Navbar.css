.logo {
  width: 90px;
  height: 70px;
}

nav {
  display: flex;
  align-items: center;
  padding: 20px 0 20px 100px;
  background-color: blue;
  color: white;
}

.home-link {
  text-decoration: none;
  color: white;
  font-weight: bold;
  font-family: sans-serif;
}

.home-link:hover {
  color: black;
}
