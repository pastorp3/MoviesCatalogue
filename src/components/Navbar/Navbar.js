import { Link } from 'react-router-dom';
import logo from '../../assets/logo.png';
import './Navbar.css';

const Navbar = () => (
  (
    <nav data-test="navComponent">
      <img data-test="logo" alt="Logo" src={logo} className="logo" />
      <Link to="/" className="home-link" data-test="homeLink">Home</Link>
    </nav>
  )
);

export default Navbar;
