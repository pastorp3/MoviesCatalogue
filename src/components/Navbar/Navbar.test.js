import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Navbar from './Navbar';
import { findByTestAtrr } from '../../../Utils';

const setUp = (props={}) => {
  const component = shallow(<Navbar {...props} />);
  return component;
};

describe('Navbar component', () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it('Should render whithout errors', () => {
    const wrapper = findByTestAtrr(component, 'navComponent');
    expect(wrapper.length).toBe(1);
  });

  it('Should render the logo', () => {
    const wrapper = findByTestAtrr(component, 'logo');
    expect(wrapper.length).toBe(1);
  });

  it('Should rnder a home link', () => {
    const wrapper = findByTestAtrr(component, 'homeLink');
    expect(wrapper.length).toBe(1);
  })

  it('Should render according with snapshout whithout error', () => {
    const tree = renderer.create(<BrowserRouter><Navbar /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
})
