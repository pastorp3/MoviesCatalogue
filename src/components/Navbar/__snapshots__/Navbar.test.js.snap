// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Should render according with snapshout whithout error 1`] = `
<nav
  data-test="navComponent"
>
  <img
    alt="Logo"
    className="logo"
    data-test="logo"
    src="logo.png"
  />
  <a
    className="home-link"
    data-test="homeLink"
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
</nav>
`;
